src/routes/docs/tutorials/pagination/step-7/+page.markdoc

```diff
@@ -0,0 +1,265 @@
+---
+layout: tutorial
+title: Implementing the offset pagination
+description: Explore offset pagination and how the react app accesses the Appwrite backend using this method.
+step: 7
+---
+In this step, we'll delve into adding the offset pagination method to our app.
+
+# What is offset pagination? {% #what-is-offset-pagination %}
+Offset pagination is widely used in databases and online APIs to extract specific data subsets from a bigger whole. It requires you to tell it where to start(offset) and how many records to get(limit). In the context of software applications, offset pagination is a technique that may be employed to facilitate the content navigation by incorporating forward and back buttons on a given page.
+
+With Appwrite, offset pagination is achieved using the `Query.limit()` and `Query.offset()` methods. Here's an example using Appwrite's `listDocuments()` function to list the next four tasks:
+
+```js
+const page2 = await databases.listDocuments(
+    '[TODOS_DATABASE_ID]',
+    '[TODOS_COLLECTION_ID]',
+    [
+        Query.limit(4),
+        Query.offset(4)
+    ]
+);
+```
+this will retrieve the next four tasks, starting at task-5
+
+We'll explore different UI patterns for offset pagination and how to implement them in our React app.
+
+# Creating the Pagination Component {% #creating-the-pagination-component %}
+{% tabs %}
+{% tabsitem #Basic-pagination title="Basic Pagination" %}
+
+Create a file and named `src/components/BasicPagination.jsx`, insert the following code:
+```js
+import React from 'react';
+
+function BasicPagination(props) {
+  const { loadPrev, loadNext, hasNext, hasPrev } = props;
+
+  return (
+    <div>
+      <button disabled={!hasPrev} onClick={loadPrev}>Previous</button>
+      <button disabled={!hasNext} onClick={loadNext}>Next</button>
+    </div>
+  );
+}
+
+export default BasicPagination;
+
+```
+This component will only have "Previous" and "Next" buttons.
+{% /tabsitem %}
+{% tabsitem #numbered-pagination title="Numbered Pagination" %}
+Create a file and named `src/components/NumberedPagination.jsx`, insert the following code:
+```js
+import React from 'react';
+
+function NumberedPagination(props) {
+  const {
+    currentPage,
+    totalPages,
+    loadPrev,
+    loadNext,
+    loadPage,
+    hasNext,
+    hasPrev
+  } = props;
+
+   const pages = [];
+
+  for (let i = 1; i <= totalPages; i++) {
+    pages.push(
+      <button key={i} disabled={i === currentPage} onClick={() => loadPage(i)}>
+        {i}
+      </button>,
+    );
+  }
+
+  return (
+    <div>
+      <button disabled={!hasPrev} onClick={loadPrev}>Previous</button>
+      {pages
+      <button disabled={!hasNext} onClick={loadNext}>Next</button>
+    </div>
+  );
+}
+
+export default NumberedPagination;
```

Comment:
https://github.com/appwrite/website/assets/29069505/9d282176-6ab9-490e-a049-f467a71f5ad5

Upon refresh, I see this....

A loop somewhere going nuts?

Reply:
oops, thats when the `numberedPagination` component gets `count.totalPages` for the first time. I'll fix it so  totalPages is its own state and has `0` for its initial state.